### DISCUSSION TITLE
CORS policy error for Vuejs + Laravel + sanctrum

### DISCUSSION THREAD

**Discussion by anwarx4u** [spatie/ray#605] [Q&A]
Hi,

Ray is working smoothly with Laravel & wordpress.
Now, I want to use ray for Vue2 (Laravel +  Vue2 +, Sancturm). But I am getting following console error.

![image](https://user-images.githubusercontent.com/21153921/143620105-64870783-4dfc-49ce-b5f0-b8124c37ec2e.png)

I think, it is not error of ray core.. But I need help to solve this.

Thanks in advance.

**Comment by anwarx4u** [comment]
@freekmurze 

can you help me regarding this issue.

**Reply by patinthehat** [reply]
@anwardote Until they have a chance to add CORS headers to the express server that runs inside Ray, you could use [ray-proxy](https://github.com/permafrost-dev/ray-proxy) as a workaround since it does provide CORS headers.

**Reply by freekmurze** [reply]
My colleage @AdrianMrn will add those headers soon in Ray itself 👍

**Comment by AdrianMrn** [comment]
@patinthehat @anwardote any idea what headers are needed? As far as I know, CORS checking is already disabled in Ray (`Access-Control-Allow-Origin = '*'`).

If I read the error correctly, the problem is that `withCredentials` is set in the original request? @anwardote could you try setting `withCredentials` in your Axios instance (or whatever you use to send the request) to `false`, and seeing if this fixes the issue?

**Reply by anwarx4u** [reply]
![image](https://user-images.githubusercontent.com/21153921/148986852-f319ce14-383f-4255-ab2a-5422c1a2b612.png)

Ya.. when I set withCredentials = false, then It works as expected.
But In my case, I need to set withCredentials=true

**Reply by AdrianMrn** [reply]
The `withCredentials` header should be set to `false` in the client then, not in the server (app). You can overwrite any headers set on the global Axios instance when you're sending a request (the one sent to Ray), without altering the instance.

Could you show which package you're using to send the logs to Ray?

**Reply by anwarx4u** [reply]
I am using Ray paid version
        "vue-ray": "^1.17.3", 
        "spatie/laravel-ray": "^1.26"

**Reply by AdrianMrn** [reply]
@patinthehat Do you think this could solve the issue downstream? https://github.com/permafrost-dev/node-ray/pull/49

A more permanent fix for this could be to create a new Axios instance that only Ray uses, so no other global config values are used.

**Reply by patinthehat** [reply]
@AdrianMrn Nice one! It's definitely worth a shot.  I'll merge and release sometime today.

**Reply by AdrianMrn** [reply]
Thanks!

**Reply by patinthehat** [reply]
@anwardote I have merged and released a PR by @AdrianMrn that may resolve this issue.  See [node-ray v1.18.0](https://github.com/permafrost-dev/node-ray/releases/tag/v1.18.0).

**Reply by anwarx4u** [reply]
Thank you so much.. it is working as expected! :-d

**Reply by AdrianMrn** [reply]
Awesome 👌
